### ISSUE TITLE
Running evaluations

### ISSUE THREAD

**Issue by BillChan226** [USC-GVL/Agent-Driver#3]
It seems that we are not able to collect the motion planning results by running `sh scripts/run_finetune.sh`, do you mean `sh scripts/run_inference.sh` instead? However, it seems that only the PlanningAgent has been instantiated?

And could you please provide some instructions on how to get the `pred_trajs_dict.pkl` in the `experiments` folder so that I could run some evaluations. Thanks!

**Comment by BillChan226** [comment]
Also how to run visualization? It seems we need to download the [nuScenes](https://www.nuscenes.org/nuscenes#download) dataset first? Could you please indicate which file should we download as well?

**Comment by BillChan226** [comment]
As I am looking through the code, it seems that the function calls are directly reading from the pickle files saved in the validation set. Thus I'm wondering wether the corresponding saved data (e.g. detected bounding boxes, predicted trajectories) is output from your pre-trained neural models or from the ground-truth labels from nuScenes? 

If the data is from the pre-trained neural models, I'd appreciate it if you could publish them as well so that the process is more transparent and easier for us to assess the agent. If not, some instructions on how to train these neural models (as claimed in the paper) would also be great to help me reproduce your result! Thanks!!

**Comment by PointsCoder** [comment]
Thanks for your interest in our work.

> It seems that we are not able to collect the motion planning results by running sh scripts/run_finetune.sh, do you mean sh scripts/run_inference.sh instead? However, it seems that only the PlanningAgent has been instantiated?

Sorry for the typo. You can run `sh scripts/run_inference.sh`. Here we used the cached outputs (data_samples.json) from perception, memory, and reasoning agent, which is mainly for reducing inference time and cost. You can also try the function [here](https://github.com/USC-GVL/Agent-Driver/blob/ab47194c8c8316d9b4c4970bde498bf89958bad0/agentdriver/main/language_agent.py#L93) without specifying the data sample, which will run the whole system end-to-end.

> And could you please provide some instructions on how to get the pred_trajs_dict.pkl in the experiments folder so that I could run some evaluations. Thanks!

I have uploaded a pred_trajs_dict.pkl for your reference.

 > Also how to run visualization? It seems we need to download the [nuScenes](https://www.nuscenes.org/nuscenes#download) dataset first? Could you please indicate which file should we download as well?

Yes. Visualization needs raw images from nuScences, so you have to download the nuScenes dataset first.

>As I am looking through the code, it seems that the function calls are directly reading from the pickle files saved in the validation set. Thus I'm wondering wether the corresponding saved data (e.g. detected bounding boxes, predicted trajectories) is output from your pre-trained neural models or from the ground-truth labels from nuScenes?

We cached the perception results from pretrained uniad modules.

 >If the data is from the pre-trained neural models, I'd appreciate it if you could publish them as well so that the process is more transparent and easier for us to assess the agent. If not, some instructions on how to train these neural models (as claimed in the paper) would also be great to help me reproduce your result! Thanks!!

The data caching process is deeply integrated into the uniad repo, so it is hard to move the code into this repo. If you'd like to reproduce the data caching part, you can simply load the pretrained checkpoint from the uniad model and hook the detection/prediction/map/occ results from its intermediate outputs.

Hope these will help.

**Comment by PointsCoder** [comment]
closed if no other questions
